scripts/bfb-install

```diff
@@ -27,19 +27,553 @@
 # of the authors and should not be interpreted as representing official policies,
 # either expressed or implied, of the FreeBSD Project.
 
+# DEBUG=1    # Define this to enable debugging output
+HOST_PIPE="/tmp/mypipe"         # Must be defined same in pipe_to_rshim.sh
+
+# Function to echo a message with color
+echo_color() {
+    local message=$1
+    # Check if the message contains certain keywords and set the color accordingly
+    if echo "$message" | grep -iq "error\|failed"; then
+        tput setaf 1;  # red color
+    elif echo "$message" | grep -iq "success\|OK"; then
+        tput setaf 2;  # green color
+    else
+        tput sgr0;  # default color
+    fi
+    echo "$message"
+    tput sgr0  # reset the color
+}
+alias echo='echo_color'
+
 usage ()
 {
-  echo "syntax: bfb-install --bfb|-b <BFBFILE> [--config|-c <bf.cfg>] \\"
-  echo "  [--rootfs|-f <rootfs.tar.xz>] --rshim|-r <rshimN> [--help|-h]"
+  cat << EOF
+Usage:
+       bfb-install  [--method  server_rshim]  --rshim  rshim_device_name  --bfb  bfb_file [--config config_file]
+       [--rootfs rootfs_file]
+
+       bfb-install --method bmc_rshim_scp --bmc bmc_host --rshim rshim_device_name --bfb bfb_file [--config con‐
+       fig_file] [--rootfs rootfs_file]
+
+       bfb-install  --method  bmc_rshim_nc  --bmc  bmc_host [--port port_number] --rshim rshim_device_name --bfb
+       bfb_file [--config config_file] [--rootfs rootfs_file]
+
+       bfb-install --method bmc_rshim_ncpipe --bmc bmc_host [--port port_number] --rshim rshim_device_name --bfb
+       bfb_file [--config config_file] [--rootfs rootfs_file]
+
+       bfb-install --help
+
+Examples:
+       To install a BFB image using the server rshim method:
+              bfb-install --rshim rshim0 --bfb bluefield.bfb
+
+       To install a BFB image using the BMC rshim scp method:
+              bfb-install --method bmc_rshim_scp --bmc lab105bmc --rshim rshim0 --bfb bluefield.bfb
+
+       To install a BFB image using the BMC rshim nc method:
```

Comment:
Same comments as above. In the following change, we will have "local rshim" and "remote rshim", instead of "host rshim"("server rshim") and "BMC rshim". BMC won't be explicitly specified unless necessary.